Define eventfd on NetBSD

krobelus opened the pull request:
Like FreeBSD, NetBSD supports eventfd, see
https://man.netbsd.org/eventfd.2.
OpenBSD does not AFAICT.

(for some reason my local rustfmt wants to reformat imports)

tgross35 commented:
Please include permalinks to the headers too, manpages don't show the values. In this case: https://github.com/NetBSD/src/blob/d04b0c735abc997743bb3faa74464524cbe7becd/sys/sys/eventfd.h.

To fix the CI failure you'll need to add the relevant header to `libc-test/build.rs`.

Why is the arg name changed in all the other files? That should be dropped, or at least moved to a separate commit if there is a reason.

krobelus commented:
split commits and updated tests and commit messages.
I'm not sure if this works for 100% of all NetBSD 10 systems out there,
but this was introduced in 2021 so it might be fine.
